cluster-autoscaler/debuggingsnapshot/debugging_snapshot.go

```diff
@@ -73,31 +77,48 @@ func (s *DebuggingSnapshotImpl) SetUnscheduledPodsCanBeScheduled(podList []*v1.P
 
 	s.UnscheduledPodsCanBeScheduled = nil
 	for _, pod := range podList {
-		s.UnscheduledPodsCanBeScheduled = append(s.UnscheduledPodsCanBeScheduled, pod)
+		s.UnscheduledPodsCanBeScheduled = append(s.UnscheduledPodsCanBeScheduled, pod.DeepCopy())
 	}
 }
 
-// SetNodeGroupInfo is the setter for Node Group Info
+// SetTemplateNodes is the setter for TemplateNodes
+func (s *DebuggingSnapshotImpl) SetTemplateNodes(templates map[string]*framework.NodeInfo) {
+	if templates == nil {
+		return
+	}
+
+	s.TemplateNodes = make(map[string]*ClusterNode)
+	for ng, template := range templates {
+		s.TemplateNodes[ng] = GetClusterNodeCopy(template)
+	}
+}
+
+// GetClusterNodeCopy is an util func to copy template node and filter values
+func GetClusterNodeCopy(template *framework.NodeInfo) *ClusterNode {
+	cNode := &ClusterNode{}
+	cNode.Node = template.Node().DeepCopy()
+	var pods []*framework.PodInfo
+	for _, p := range template.Pods {
+		pods = append(pods, p.DeepCopy())
```

Comment:
I just realized this is PodInfo, which is another scheduler wrapper (similarly to how NodeInfo wraps Node). Maybe we should consider switching to PodInfo.Pod.DeepCopy()? I'm pretty sure we don't need anything in PodInfo other than the actual pod.

Reply:
Done.